src/views/Tasks/ViewTask/index.jsx

```diff
@@ -43,20 +69,35 @@ import pageArtifactsQuery from './pageArtifacts.graphql';
   }),
 })
 export default class ViewTask extends Component {
-  static getDerivedStateFromProps(nextProps, prevState) {
-    return {
-      taskSearch:
-        nextProps.match.params.taskId !== prevState.taskSearch
-          ? nextProps.match.params.taskId || ''
-          : prevState.taskSearch,
-      showError: !!nextProps.data.error,
+  constructor(props) {
+    super(props);
+
+    const taskId = props.match.params.taskId || '';
+
+    this.state = {
+      taskSearch: taskId,
+      showError: false,
     };
+
+    if (taskId) {
+      updateTaskIdHistory(taskId);
+    }
   }
 
-  state = {
-    taskSearch: '',
-    showError: false,
-  };
+  componentDidUpdate(prevProps) {
+    const { taskId } = this.props.match.params;
+
+    if (prevProps.match.params.taskId !== taskId) {
+      updateTaskIdHistory(taskId);
+      this.props.history.push(`/tasks/${taskId}`);
```

Comment:
Not sure why this push is here; this should already be redirected to if `this.props.match.params` has changed.